Ratios/CalculatorViewController.swift

```diff
@@ -8,6 +8,55 @@
 
 import UIKit
 
+fileprivate struct RatioColour {
+    static let strongBackgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 1)
+    static let weakBackgroundColor   = UIColor(red: 1, green: 1, blue: 1, alpha: 1)
+
+    static let minRatio: Int = 12
+    static let maxRatio: Int = 20
+
+    let value: Int
+    let colour: UIColor
+    
+    private static func colorFor(ratio: Int) -> UIColor {
+        var value = ratio
+        
+        if value > maxRatio {
+            value = maxRatio
+        } else if value < minRatio {
+            value = minRatio
+        }
+        
+        let color_offset_fraction = CGFloat(Double(value - minRatio) / Double(maxRatio - minRatio))
+        
+        var minRed: CGFloat = 0,    maxRed: CGFloat = 0
+        var minGreen: CGFloat = 0,  maxGreen: CGFloat = 0
+        var minBlue: CGFloat = 0,   maxBlue: CGFloat = 0
+        
+        strongBackgroundColor.getRed(&minRed, green: &minGreen, blue: &minBlue, alpha: nil)
+        weakBackgroundColor.getRed(&maxRed, green: &maxGreen, blue: &maxBlue, alpha: nil)
+        
+        let get_new_color = { (min: CGFloat, max: CGFloat) -> CGFloat in
```

Comment:
Can this be made a private static function on `RatioColour`?

Reply:
I opted for a closure because the logic here is pretty tightly coupled to the the function context. Since Swift doesn't have nested functions, I thought it made sense to keep it close to its usage context.

Let me know if you still want the change, though